services/src/modules/plugins/metrics.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  ApolloServerPlugin,
+  GraphQLRequestContextWillSendResponse,
+  GraphQLFieldResolverParams,
+} from 'apollo-server-plugin-base';
+import { FastifyInstance } from 'fastify';
+import { Histogram } from 'prom-client';
+
+let requestDurationHistogram: Histogram<string> | undefined;
+let resolverDurationHistogram: Histogram<string> | undefined;
+
+export function createMetricsPlugin(fastifyInstance: Pick<FastifyInstance, 'metrics'>): ApolloServerPlugin {
```

Comment:
If all we need is `metrics.client`, maybe just pass that bit in? Then the metrics plugin only depends on `prom-client`, not on fastify itself

Reply:
The problem that this client isn't created when the plugins are defined. This is the reason I send the instance itself